src/07-reference.md

```diff
@@ -0,0 +1,580 @@
+# Language reference
+
+This is a living specification of the current set of built-ins in the `.lurk` package.
+
+## Built-in atoms
+
+### `nil`
+
+`nil` is the empty list and used as boolean false. 
+
+It has its own type differently from other symbols.
+
+```
+user> (eq nil '())
+[3 iterations] => t
+user> (cdr '(1))
+[2 iterations] => nil
+user> (if nil "true" "false")
+[3 iterations] => "false"
+```
+
+### `t`
+
+`t` is boolean true. In boolean contexts like `if`, anything that isn't `nil` is considered false, but `t` is generally used.
+
+```
+user> (eq 1 1)
+[2 iterations] => t
+user> (if t "true" "false")
+[3 iterations] => "true"
+```
+
+## Built-in functions
+
+### `atom`
+
+`(atom x)` returns `nil` if `x` is a list or a pair, and `t` otherwise.
```

Comment:
It's probably better to just say "pair" because `nil` is the empty list.

If we try to treat lists as first class objects, the details won't match. It's better to treat lists as constructions on top of pairs, like I did in the Lisp chapter. Then everything fits nicely.